syntax ="proto3";

enum Commands {
  STOP = 0;
  RUN = 1;
  BREAK = 2;
}

enum Kicker {
  NO_KICK = 0;
  KICK1 = 1;
  KICK2 = 2;
}

message MainBoardToBrushless {
  Commands command = 1; // Brushless command
  float speed = 2; // m.s-1
}

message BrushlessToMainBoard {
  uint32 error_count = 1; // Number of SPI transmission errors
  float measured_speed = 2; // m.s-1
}

message IAToMainBoard {
  uint32 robot_id = 1; // Robot identifier
  float normal_speed = 2; // m.s-1
  float tangential_speed = 3; // m.s-1
  float angular_speed = 4; // rad.s-1
  bool motor_break = 5; // Break
  Kicker kicker_cmd = 6; // dont kick, kick with Kicker 1 or Kicker 2
  float kick_power = 7; // kick power (uS)
  bool charge = 8; // enable / disable charge kicker
  bool dribbler = 9; // Enable / disable dribbler
}

message MainboardToIA {
  uint32 robot_id = 1; // Robot identifier
  float measured_normal_speed = 2; // m.s-1
  float measured_tangential_speed = 3; // m.s-1
  float measured_angular_speed = 4; // rad.s-1
}
